consensus/consensus-types/src/block_info.rs

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) The Libra Core Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::block::Block;
+use crypto::hash::HashValue;
+use failure::prelude::{Error, Result};
+use libra_types::transaction::Version;
+use serde::{Deserialize, Serialize};
+use std::convert::TryFrom;
+
+/// The round of a block is a consensus-internal counter, which starts with 0 and increases
+/// monotonically. It is used for the protocol safety and liveness (please see the detailed protocol description).
+pub type Round = u64;
+
+/// This structure contains all the information needed for tracking a block
+/// without having access to the block or its execution output state. It
+/// assumes that the block is the last block executed within the ledger.
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+pub struct BlockInfo {
+    /// Epoch number corresponds to the set of validators that are active for this block.
+    epoch_num: u64,
+    /// The consensus protocol executes proposals (blocks) in rounds, which monotically increase per epoch.
+    round: Round,
+    /// The identifier (hash) of the block.
+    id: HashValue,
+    /// The accumulator root hash after executing this block.
+    executed_state_id: HashValue,
+    /// The version of the latest transaction in the ledger.
+    version: Version,
+    /// The timestamp this block was proposed by a proposer.
+    timestamp_usecs: u64,
+}
+
+impl BlockInfo {
+    pub fn new(
+        epoch_num: u64,
+        round: Round,
+        id: HashValue,
+        executed_state_id: HashValue,
+        version: Version,
+        timestamp_usecs: u64,
+    ) -> Self {
+        Self {
+            epoch_num,
+            round,
+            id,
+            executed_state_id,
+            timestamp_usecs,
+            version,
+        }
+    }
+
+    pub fn empty() -> Self {
+        Self {
+            epoch_num: 0,
+            round: 0,
+            id: HashValue::zero(),
+            executed_state_id: HashValue::zero(),
+            version: 0,
+            timestamp_usecs: 0,
+        }
+    }
+
+    pub fn from_block<T>(block: &Block<T>, executed_state_id: HashValue, version: Version) -> Self {
```

Comment:
it'd look nicer if we have ExecutedBlock in consensus_types so it's just `from_block<T>(block: &ExecutedBlock<T>) -> Self`